corehq/motech/value_source.py

```diff
@@ -126,7 +126,11 @@ class CaseProperty(ValueSource):
     case_property = StringProperty()
 
     def _get_commcare_value(self, case_trigger_info):
-        return case_trigger_info.updates.get(self.case_property)
+        case_property_values = dict(
+            list(case_trigger_info.extra_fields.items()) +
+            list(case_trigger_info.updates.items())
+        )
+        return case_property_values.get(self.case_property)
```

Comment:
So update takes precedence over extra_fields in case a case_property is present in both?
how about just
```
if self.case_property in case_trigger_info.updates:
    return case_trigger_info.updates[self.case_property]
return case_trigger_info.extra_fields.get(self.case_property)
```